paasta_tools/autoscaling/autoscaling_service_lib.py

```diff
@@ -353,8 +359,20 @@ def autoscale_marathon_instance(marathon_service_config, marathon_tasks, mesos_t
         **autoscaling_params
     )
 
-    new_instance_count = marathon_service_config.limit_instance_count(current_instances + autoscaling_amount)
+    # Limit downscaling by 30% of current_instances until we find out what is
+    # going on in such situations
+    safe_downscaling_threshold = int(current_instances * 0.7)
+    new_instance_count = max(current_instances + autoscaling_amount, safe_downscaling_threshold)
+
+    new_instance_count = marathon_service_config.limit_instance_count(new_instance_count)
     if new_instance_count != current_instances:
+        if new_instance_count == safe_downscaling_threshold:
+            write_to_log(
+                config=marathon_service_config,
+                line='Autoscaler clamped: %s' % str(log_utilization_data),
```

Comment:
It isn't returned, it is supposed to be passed by reference from autoscale_marathon_instance(): https://github.com/Yelp/paasta/pull/1000/files#diff-699ef1f9441fd1dab1f660ff54347041R334

Each metrics_provider returns a number, and I wanted to keep it this way, otherwise I would have to adjust all of them.